src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Move.cs

```diff
@@ -92,6 +126,74 @@ private void DirectoryMove_DifferentWatchedDirectory(WatcherChangeTypes eventTyp
             }
         }
 
+        private void DirectoryMove_Multiple_FromWatchedToUnwatched(int filesCount, bool skipOldEvents)
+        {
+            Assert.True(filesCount > 0);
+
+            using (var watchedTestDirectory = new TempDirectory(GetTestFilePath()))
+            using (var unwatchedTestDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var dirs = Enumerable.Range(0, filesCount)
+                                .Select(i => new
+                                {
+                                    DirecoryInWatchedDir = Path.Combine(watchedTestDirectory.Path, $"dir{i}"),
+                                    DirecoryInUnwatchedDir = Path.Combine(unwatchedTestDirectory.Path, $"dir{i}")
+                                }).ToArray();
+
+                Array.ForEach(dirs, (dir) => Directory.CreateDirectory(dir.DirecoryInWatchedDir));
+
+                using (var watcher = new FileSystemWatcher(watchedTestDirectory.Path, "*"))
+                {
+                    Action action = () => Array.ForEach(dirs, dir => Directory.Move(dir.DirecoryInWatchedDir, dir.DirecoryInUnwatchedDir));
+
+                    // On macOS, for each file we receive two events as describe in comment below.
+                    int expectEvents = filesCount;
+                    if (skipOldEvents)
+                        expectEvents = expectEvents * 2;
+
+                    IEnumerable<FiredEvent> events = ExpectEvents(watcher, expectEvents, action);
+
+                    if (skipOldEvents)
+                        events = events.Where(x => x.EventType != WatcherChangeTypes.Created);
+
+                    var expectedEvents = dirs.Select(dir => new FiredEvent(WatcherChangeTypes.Deleted, dir.DirecoryInWatchedDir));
+
+                    // Remove Created events as there is racecondition when create dir and then observe parent folder. It receives Create event altought Watcher is not registered yet.
+                    Assert.Equal(expectedEvents, events.Where(x => x.EventType != WatcherChangeTypes.Created));
+                }
+            }
+        }
+
+        private void DirectoryMove_Multiple_FromUnwatchedToWatched(int filesCount)
+        {
+            Assert.True(filesCount > 0);
+
+            using (var watchedTestDirectory = new TempDirectory(GetTestFilePath()))
+            using (var unwatchedTestDirectory = new TempDirectory(GetTestFilePath()))
```

Comment:
nit: consider a using declaration to avoid double-indenting the entire method body.